File: Suhyun/Level2/타겟넘버/TargetNumber.java
package Level2.타겟넘버;

public class TargetNumber {
    int[] arr;
    int answer = 0;

    public int main(String[] args) {
        int[] numbers = { 1, 1, 1, 1, 1 };
        int target = 3;

        int[] arr;
        int answer = 0;

        arr = numbers;

        dfs(0, 0, target);

        // return answer;

    }

    public void dfs(int sum, int i, int target) {
        if (i == arr.length) {
            if (sum == target)
                answer++;
            return;
        }

        dfs(sum + arr[i], i + 1, target);
        dfs(sum - arr[i], i + 1, target);
    }
}

File: Suhyun/Level2/타겟넘버/TargetNumber.js
function solution(numbers, target) {
    let answer = 0;

    function dfs(depth, sum) {
        if (depth === numbers.length) {
            if (sum === target) {
                answer += 1;
            }
            return;
        }
        dfs(depth + 1, sum + numbers[depth]);
        dfs(depth + 1, sum - numbers[depth]);
    }

    dfs(0, 0);

    return answer;
}